### PR TITLE
feat(BILL-CPC-1055): add update bill for admin

### PR DESCRIPTION
**Pull request by PaoloOabel**
JIRA: https://champlainsaintlambert.atlassian.net/browse/CPC-1055
## Context:

Implemented update bill feature to allow admins to update a feature.

## Does this PR change the .vscode folder in petclinic-frontend?:
No it does not

## Changes

- Implemented update method in BillController in the api-gateway
- Implemented a new update form page
- Implemented API request to update in the frontend
- Implemented 2 tests for the controller method (one positive, one negative)

## Before and After UI (Required for UI-impacting PRs)
Before
beforeUpdateBill

After:
afterUpdateBill1
afterUpdateBill2
afterUpdateBill3

### CODE REVIEW

**Review by ctsaki25 — DISMISSED** [review]
Everything looks good. The only issue I see is with the package-lock.json and package.json, can you keep the original files? You can also add more tests in the future.

File: package.json
Comment (Futurespast) [inline package.json:1]: All the package-lock changes are spacing. So I think it's fine

**Review by Futurespast — APPROVED** [review]

File: package.json
Comment (PaoloOabel) [reply package.json:1]: I restored the original ones in the end, no changes are done to them.

**Comment by ctsaki25** [comment]
Be sure to run 'npm run lint -- --fix' !

**Review by Futurespast — APPROVED** [review]
Nice changes, glad you were able to fix everything.

File: petclinic-frontend/src/features/bills/AdminBillsListTable.tsx [hunk 1]
```diff
@@ -130,14 +132,17 @@ export default function AdminBillsListTable(): JSX.Element {
     try {
       const response = await deleteBill(billToDelete);
       if (response.status === 200 || response.status === 204) {
-        window.alert('Bill (${billId}) has been deleted successfully');
+        window.alert('Bill ${billId} has been deleted successfully');
```
Comment (ctsaki25) [inline petclinic-frontend/src/features/bills/AdminBillsListTable.tsx:135]: glad you caught this, I don't know how this happened.

**Review by ctsaki25 — APPROVED** [review]
Everything looks great. Excellent work. Im glad you were able to fix all issues and catch my mistake

**Pull request merged**